suite/tests/client-interface/drbbdup-emul-reg-clobber-test.dll.c

```diff
@@ -0,0 +1,311 @@
+/* **********************************************************
+ * Copyright (c) 2015-2022 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* Tests the drbbdup extension's interactions with emulation. */
+
+#include "dr_api.h"
+#include "client_tools.h"
+#include "drmgr.h"
+#include "drbbdup.h"
+#include "drutil.h"
+#include "drreg.h"
+#include <string.h>
+
+typedef struct _per_block_t {
+    bool saw_first;
+    bool saw_last;
+} per_block_t;
+
+/* Assume single threaded. */
+static uintptr_t encode_val = 1;
+
+#ifdef X86
+static bool saw_movs;
+#endif
+static bool has_rest_of_block_emul;
+
+static dr_emit_flags_t
+app2app_event(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
+              bool translating)
+{
+#if VERBOSE
+    /* Useful for debugging. */
+    instrlist_disassemble(drcontext, tag, bb, STDERR);
+#endif
+
+    has_rest_of_block_emul = false;
+
+    /* Test drutil rep string expansion interacting with drbbdup. */
+    bool expanded;
+    if (!drutil_expand_rep_string_ex(drcontext, bb, &expanded, NULL))
+        DR_ASSERT(false);
+    if (expanded) {
+        /* We can't overlap ours with drutil's so bow out. */
+        has_rest_of_block_emul = true;
+        return DR_EMIT_DEFAULT;
+    }
+    /* Test handling of DR_EMULATE_REST_OF_BLOCK by inserting it in the middle
```

Comment:
This is all an unmodified copy from drbbdup-emul-test: but there is no need to duplicate all the checks done in that test here, right?  Better to remove everything except what's needed to test the issue at hand: then we'll have a clearer, more focused test.  As it is, a reader will look at this and have no idea what's different vs the other test: there aren't even any comments saying how it's different.

Reply:
Removed these lines.